cmake/test.cmake

```diff
@@ -165,22 +166,30 @@ endfunction()
 #
 # if C++ standard <version_number> is supported by the compiler and the
 # source file contains JSON_HAS_CPP_<version_number>.
+# Use NAME <name> to override the filename-derived test name.
```

Comment:
This new argument to `json_test_add_test_for()` is used to add the same unit test file multiple times with different build settings.